packages/static-build/src/index.ts

```diff
@@ -637,6 +632,30 @@ export const build: BuildV2 = async ({
 
       const outputDirPrefix = path.join(workPath, path.dirname(entrypoint));
 
+      // If the Build Command or Framework output files according to the
+      // Build Output v3 API, then stop processing here in `static-build`
+      // since the output is already in its final form.
+      const buildOutputPath = await BuildOutputV3.getBuildOutputDirectory(
+        outputDirPrefix
+      );
+
+      if (buildOutputPath) {
+        // Ensure that `vercel build` is being used for this Deployment
+        if (!meta.cliVersion) {
+          let buildCommandName: string;
+          if (buildCommand) buildCommandName = `"${buildCommand}"`;
+          else if (framework) buildCommandName = framework.name;
+          else buildCommandName = 'the "build" script';
+          throw new Error(
+            `Detected Build Output v3 from ${buildCommandName}, but this Deployment is not using \`vercel build\`.\nPlease set the \`ENABLE_VC_BUILD=1\` environment variable. More info: ____________`
```

Comment:
Will we link this to a card somewhere so we don't forget to update it? I think we should remove the `More info` part until we have that link ready.